src/sql/src/plan/statement/ddl.rs

```diff
@@ -1090,8 +1103,20 @@ pub fn plan_create_subsource(
         columns,
         constraints,
         if_not_exists,
+        with_options,
     } = &stmt;
 
+    let CreateSubsourceOptionExtracted {
+        progress,
+        references,
+        ..
+    } = with_options.clone().try_into()?;
+
+    assert!(
+        progress ^ references,
```

Comment:
This isn't enforced in the parser, so I think this would make it very easy for a user to panic the server. Can this be an error instead?

Edit: oh it's enforced by the purifier. Can you add a comment about that here, and also add a test somewhere for it?

Reply:
This is implcitly tested on every `CREATE SOURCE` statement that generates subsources, but to make that clearer I've added comments.